Tools/build/check_warnings.py

```diff
@@ -78,9 +122,24 @@ def get_warnings_by_file(warnings: list[dict]) -> dict[str, list[dict]]:
     return warnings_by_file
 
 
+def is_file_ignored(
+    file_path: str, ignore_rules: set[IgnoreRule]
+) -> IgnoreRule | None:
+    """
+    Returns the IgnoreRule object for the file path if there is a related rule for it
+    """
+    for ignore_rule in ignore_rules:
+        if ignore_rule.is_directory:
+            if file_path.startswith(ignore_rule.file_path):
+                return ignore_rule
+        elif file_path == ignore_rule.file_path:
+            return ignore_rule
```

Comment:
Let's follow an example below, just call it `rule`, we only have one kind of rule and makes it a bit easier to read:
```suggestion
    for rule in ignore_rules:
        if rule.is_directory:
            if file_path.startswith(rule.file_path):
                return rule
        elif file_path == rule.file_path:
            return rule
```

Reply:
@hugovk I agree. renamed to `rule`

Reply:
or did you mean every instance of `ignore_rule` change to `rule`

Reply:
👍 all instances of an IgnoreRule is a `rule`